portfolio/src/main/java/com/google/sps/data/WordCountUpdater.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
+
+public class WordCountUpdater{
+  
+  private static final int TRANSACTION_RETRIES = 4;
+  private DatastoreService datastore;
+  
+  public WordCountUpdater() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+  
+  public void updateWordCount(String comment, char op) {
+    String[] words = comment.replaceAll("[^a-zA-Z ]", "").toLowerCase().split("\\s+");
+    
+    for(String word : words){
+      Entity wordEntity;
+      Key wordKey = KeyFactory.createKey("word", word + "_word");
+      Transaction txn = datastore.beginTransaction(TransactionOptions.Builder.withXG(true));
+      try {
+        wordEntity = datastore.get(txn, wordKey);
+      } catch (com.google.appengine.api.datastore.EntityNotFoundException e) {
+        // In case word is not yet in datastore...
+        wordEntity = new Entity(wordKey);
+        wordEntity.setProperty("count", 0L);
+      }
+      long newCount = (Long) wordEntity.getProperty("count") + (op == '+' ? 1 : -1);
```

Comment:
No space between (Long) and wordEntity.

Reply:
Tested it with the formatter - they prefer the space actually.

Reply:
Hmm, ok, I'd ask Jack then. I know in the past he's asked me to remove spaces between castors and the thing they're casting.